src/Compilers/CSharp/Test/Semantic/Semantics/StaticNullChecking.cs

```diff
@@ -14578,6 +14578,86 @@ class C : I
             Assert.Equal(new[] { "void I.M<T>(T? x)" }, implementations.SelectAsArray(m => m.ToTestDisplayString()));
         }
 
+        [Fact]
+        public void EmptyStructDifferentAssembly()
+        {
+            var sourceA =
+@"using System.Collections;
+public struct S
+{
+    public S(string f, IEnumerable g)
+    {
+        F = f;
+        G = g;
+    }
+    private string F { get; }
+    private IEnumerable G { get; }
+}";
+            var compA = CreateStandardCompilation(sourceA);
+            var sourceB =
+@"using System.Collections.Generic;
+class C
+{
+    static void Main()
+    {
+        var c = new List<object>();
+        c.Add(new S(string.Empty, new object[0]));
+    }
+}";
+            var compB = CreateStandardCompilation(
+                new[] { sourceB, attributesDefinitions },
+                options: TestOptions.ReleaseExe,
+                parseOptions: TestOptions.Regular.WithNullCheckingFeature(),
+                references: new[] { compA.EmitToImageReference() });
+            CompileAndVerify(compB, expectedOutput: "");
```

Comment:
Without the checks for `slot > 0`, the assert in `DataFlowPass.InheritNullableStateOfTrackableStruct` will fail. I believe the assert can be ignored and compilation will succeed though, at least with the tests here. Since we were checking `slot > 0` in all other cases, it seemed reasonable to add the two missing checks rather than remove the assert and require `InheritNullableStateOfTrackableStruct` to handle other cases.